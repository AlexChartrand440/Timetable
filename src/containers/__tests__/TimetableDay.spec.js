// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import moment from 'moment';
import { WEEKLY } from '../../lib/recurrence';
import { MON } from '../../lib/weekdays';
import { TimetableDay } from '../TimetableDay';

const courses: CoursesState = {
  course1: {
    id: 'course1',
    name: 'A course',
    appointments: ['app1'],
  },
  course2: {
    id: 'course2',
    name: 'Another course',
    appointments: ['app2'],
  },
};
const startdate = moment('01-01-2017', 'MM-DD-YYYY').day(0); // sunday
const enddate = startdate.add(3, 'weeks');
const appointmentsOfDay: Array<Appointment> = [
  {
    id: 'app1',
    course: 'course1',
    startdate,
    enddate,
    starttime: new Date(1),
    endtime: new Date(2),
    day: MON,
    recurrence: WEEKLY,
  },
  {
    id: 'app2',
    course: 'course2',
    startdate,
    enddate,
    starttime: new Date(1),
    endtime: new Date(2),
    day: MON,
    recurrence: WEEKLY,
  },
];

const props = {
  courses,
  day: 'mon',
  appointmentsOfDay,
  onEditCourse: jest.fn(),
};

describe('TimetableDay', () => {
  it('renders a CourseAppointment for each element of prop appointmentsOfDay', () => {
    const tree = renderer.create(<TimetableDay {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('calls props.onEditCourse(course.id) when CourseAppointment calls onEdit', () => {
    const wrapper = shallow(<TimetableDay {...props} />);
    const course = courses.course1;
    wrapper.find({ name: course.name }).simulate('edit');
    expect(props.onEditCourse).toBeCalledWith(course.id);
  });
});
