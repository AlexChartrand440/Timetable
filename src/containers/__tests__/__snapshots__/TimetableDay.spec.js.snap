// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimetableDay renders a CourseAppointment for each element of prop appointmentsOfDay 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "borderRadius": 2,
      "marginVertical": 5,
      "paddingTop": 4,
      "shadowColor": "black",
      "shadowOffset": Object {
        "height": 6,
      },
      "shadowOpacity": 0.05,
      "shadowRadius": 2,
    }
  }
>
  <View
    style={
      Object {
        "marginLeft": 10,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#797979",
          "fontFamily": "Helvetica Neue",
          "fontSize": 16,
          "fontWeight": "400",
        }
      }
    >
      mon
    </Text>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#95989A",
        "height": 0.5,
        "marginHorizontal": 3,
        "opacity": 0.4,
      }
    }
  />
  <View>
    <View
      leftActionActivationDistance={125}
      leftActionReleaseAnimationConfig={null}
      leftActionReleaseAnimationFn={null}
      leftButtonWidth={75}
      leftButtonsActivationDistance={75}
      leftButtonsCloseReleaseAnimationConfig={null}
      leftButtonsCloseReleaseAnimationFn={null}
      leftButtonsOpenReleaseAnimationConfig={null}
      leftButtonsOpenReleaseAnimationFn={null}
      onLayout={[Function]}
      onLeftActionActivate={[Function]}
      onLeftActionComplete={[Function]}
      onLeftActionDeactivate={[Function]}
      onLeftActionRelease={[Function]}
      onLeftButtonsActivate={[Function]}
      onLeftButtonsCloseComplete={[Function]}
      onLeftButtonsCloseRelease={[Function]}
      onLeftButtonsDeactivate={[Function]}
      onLeftButtonsOpenComplete={[Function]}
      onLeftButtonsOpenRelease={[Function]}
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onPanAnimatedValueRef={[Function]}
      onRef={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onRightActionActivate={[Function]}
      onRightActionComplete={[Function]}
      onRightActionDeactivate={[Function]}
      onRightActionRelease={[Function]}
      onRightButtonsActivate={[Function]}
      onRightButtonsCloseComplete={[Function]}
      onRightButtonsCloseRelease={[Function]}
      onRightButtonsDeactivate={[Function]}
      onRightButtonsOpenComplete={[Function]}
      onRightButtonsOpenRelease={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      onSwipeComplete={[Function]}
      onSwipeMove={[Function]}
      onSwipeRelease={[Function]}
      onSwipeStart={[Function]}
      rightActionActivationDistance={125}
      rightActionReleaseAnimationConfig={null}
      rightActionReleaseAnimationFn={null}
      rightButtonWidth={75}
      rightButtonsActivationDistance={75}
      rightButtonsCloseReleaseAnimationConfig={null}
      rightButtonsCloseReleaseAnimationFn={null}
      rightButtonsOpenReleaseAnimationConfig={null}
      rightButtonsOpenReleaseAnimationFn={null}
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
          undefined,
        ]
      }
      swipeReleaseAnimationConfig={
        Object {
          "duration": 250,
          "easing": [Function],
          "toValue": Object {
            "x": 0,
            "y": 0,
          },
        }
      }
      swipeReleaseAnimationFn={[Function]}
      swipeStartMinDistance={15}
    >
      <View
        style={
          Object {
            "marginLeft": -0,
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
            "width": 0,
          }
        }
      >
        <View
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
              "width": 0,
            }
          }
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hitSlop={undefined}
            isTVSelectable={true}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "flex-end",
                "backgroundColor": "#70C1B3",
                "flex": 1,
                "justifyContent": "center",
                "opacity": 1,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              onLayout={[Function]}
              style={
                Array [
                  Object {
                    "color": "white",
                    "fontFamily": "Helvetica Neue",
                    "fontSize": 16,
                    "fontWeight": "300",
                  },
                  Object {
                    "marginHorizontal": 10,
                  },
                ]
              }
            >
              Edit
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
            "zIndex": 1,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "white",
              "borderRadius": 3,
              "right": 2,
              "width": 754,
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
                "marginHorizontal": 20,
                "paddingVertical": 5,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "flex-start",
                  "flex": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                >
                  01:00
                </Text>
                <View
                  style={
                    Object {
                      "backgroundColor": "#95989A",
                      "height": 8,
                      "width": 1,
                    }
                  }
                />
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                >
                  01:00
                </Text>
              </View>
            </View>
            <View
              style={
                Object {
                  "flex": 6,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "paddingRight": 10,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#247BA0",
                    "fontFamily": "Helvetica Neue",
                    "fontSize": 20,
                    "fontWeight": "300",
                  }
                }
              >
                A course
                 
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#247BA0",
                    "fontFamily": "Helvetica Neue",
                    "fontSize": 20,
                    "fontWeight": "300",
                  }
                }
              >
                
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#95989A",
          "height": 0.5,
          "marginHorizontal": 3,
          "opacity": 0.4,
        }
      }
    />
    <View
      leftActionActivationDistance={125}
      leftActionReleaseAnimationConfig={null}
      leftActionReleaseAnimationFn={null}
      leftButtonWidth={75}
      leftButtonsActivationDistance={75}
      leftButtonsCloseReleaseAnimationConfig={null}
      leftButtonsCloseReleaseAnimationFn={null}
      leftButtonsOpenReleaseAnimationConfig={null}
      leftButtonsOpenReleaseAnimationFn={null}
      onLayout={[Function]}
      onLeftActionActivate={[Function]}
      onLeftActionComplete={[Function]}
      onLeftActionDeactivate={[Function]}
      onLeftActionRelease={[Function]}
      onLeftButtonsActivate={[Function]}
      onLeftButtonsCloseComplete={[Function]}
      onLeftButtonsCloseRelease={[Function]}
      onLeftButtonsDeactivate={[Function]}
      onLeftButtonsOpenComplete={[Function]}
      onLeftButtonsOpenRelease={[Function]}
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onPanAnimatedValueRef={[Function]}
      onRef={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onRightActionActivate={[Function]}
      onRightActionComplete={[Function]}
      onRightActionDeactivate={[Function]}
      onRightActionRelease={[Function]}
      onRightButtonsActivate={[Function]}
      onRightButtonsCloseComplete={[Function]}
      onRightButtonsCloseRelease={[Function]}
      onRightButtonsDeactivate={[Function]}
      onRightButtonsOpenComplete={[Function]}
      onRightButtonsOpenRelease={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      onSwipeComplete={[Function]}
      onSwipeMove={[Function]}
      onSwipeRelease={[Function]}
      onSwipeStart={[Function]}
      rightActionActivationDistance={125}
      rightActionReleaseAnimationConfig={null}
      rightActionReleaseAnimationFn={null}
      rightButtonWidth={75}
      rightButtonsActivationDistance={75}
      rightButtonsCloseReleaseAnimationConfig={null}
      rightButtonsCloseReleaseAnimationFn={null}
      rightButtonsOpenReleaseAnimationConfig={null}
      rightButtonsOpenReleaseAnimationFn={null}
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
          undefined,
        ]
      }
      swipeReleaseAnimationConfig={
        Object {
          "duration": 250,
          "easing": [Function],
          "toValue": Object {
            "x": 0,
            "y": 0,
          },
        }
      }
      swipeReleaseAnimationFn={[Function]}
      swipeStartMinDistance={15}
    >
      <View
        style={
          Object {
            "marginLeft": -0,
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
            "width": 0,
          }
        }
      >
        <View
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
              "width": 0,
            }
          }
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hitSlop={undefined}
            isTVSelectable={true}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "flex-end",
                "backgroundColor": "#70C1B3",
                "flex": 1,
                "justifyContent": "center",
                "opacity": 1,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              onLayout={[Function]}
              style={
                Array [
                  Object {
                    "color": "white",
                    "fontFamily": "Helvetica Neue",
                    "fontSize": 16,
                    "fontWeight": "300",
                  },
                  Object {
                    "marginHorizontal": 10,
                  },
                ]
              }
            >
              Edit
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
            "zIndex": 1,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "white",
              "borderRadius": 3,
              "right": 2,
              "width": 754,
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
                "marginHorizontal": 20,
                "paddingVertical": 5,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "flex-start",
                  "flex": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                >
                  01:00
                </Text>
                <View
                  style={
                    Object {
                      "backgroundColor": "#95989A",
                      "height": 8,
                      "width": 1,
                    }
                  }
                />
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                >
                  01:00
                </Text>
              </View>
            </View>
            <View
              style={
                Object {
                  "flex": 6,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "paddingRight": 10,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#247BA0",
                    "fontFamily": "Helvetica Neue",
                    "fontSize": 20,
                    "fontWeight": "300",
                  }
                }
              >
                Another course
                 
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#247BA0",
                    "fontFamily": "Helvetica Neue",
                    "fontSize": 20,
                    "fontWeight": "300",
                  }
                }
              >
                
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
